otter/test/test_auth.py

```diff
@@ -92,7 +88,7 @@ def test_authenticate_user(self):
         self.treq.post.return_value = succeed(response)
 
         d = authenticate_user('http://identity/v2.0', 'user', 'pass',
-                              log=self.log)
+                              log=self.log, pool=pool)
```

Comment:
Apologies, again for the bother :|  Would it be possible to test whether `pool is None`, and if so, just don't pass pool?  That way we can test that not passing a parameter won't break, as opposed to explicitly passing it as None?

This is much neater though, thank you for the refactor!!